### DISCUSSION TITLE
Asynchronous API-Requests Seem to Block

### DISCUSSION THREAD

**Discussion by fabiangfd** [mirumee/ariadne#1142] [Q&A]
Hi,

I am using Ariadne in my Django project and run it as an _ASGI-Application_ (required, since I am using _graphql-subscriptions_). 
**Accordingly, all my resolvers are _async_ functions**. Overall, Ariadne has been amazing to work with!! 

One of my mutations happens to take a few seconds to resolve.
While waiting for the mutation results, I have noticed a surprising behavior: **Any call to the API that is sent while the mutation has not yet returned, is blocked until the mutation has finished resolving**.

This is not what I had expected from an asynchronous application. Why is this happening? How can I make my API truly async?

Any help would be greatly appreciated.

**Comment by rafalp** [comment]
Sounds like you are doing a synchronous IO that blocks whole server. Are you sure you are using async ORM or wrapping ORM calls in `sync_to_async`?


Also, async is currently broken in Django:

https://code.djangoproject.com/ticket/33497
